nep-Y.mediawiki

```diff
@@ -0,0 +1,331 @@
+  Created: 2022-09-12
+</pre>
+
+==Abstract==
+
+This proposal describes some Neo-specific JSON-RPC interaction requirements
+along with a set of Neo-specific error codes to be used by compliant
+implementations. It allows different Neo nodes with different RPC server
+implementations to be compatible and able to interact with various RPC
+clients.
+
+==Motivation==
+
+Currently different implementations of RPC servers in Neo nodes use slightly
+different set of error codes and clients can't rely on codes to distinguish
+one errors from another. Some errors were also changed on update from Legacy
+to N3. The overall set is also too coarse to be used effectively by clients,
+many different problems requiring different client actions end up with the
+same code, so clients have to parse error messages to react correctly which is
+far less reliable than using codes. At the same time JSON-RPC is too
+permissive in some of its definitions, while Neo can benefit from stricter
+definitions used.
+
+A standard set of errors and JSON-RPC conventions solve these problems by
+providing common language both for servers and clients. Servers can have a
+reference on what code should be used for what purpose and what data types
+should be used, while clients can easily check for error codes and react
+accordingly.
+
+==Specification==
+
+===General JSON-RPC conventions===
+
+JSON-RPC 2.0 specification for the <code>Error</code> object is augmented with
+the following compatible clauses:
+
+* <code>code</code> field MUST be a negative integer value ranging from -1 to -65536 that is reserved for a particular error by the specification. <code>code</code> field MUST be the same for each Neo node irrespective of the implementation details except for the reserved experimental implementation-defined range. <code>code</code> field MUST NOT be omitted. 
+* <code>message</code> field MUST be a valid UTF-8 string value that specifies a short error description. <code>message</code> value MAY NOT be exactly the same across all Neo node implementations (we have <code>code</code> for this). <code>message</code> value SHOULD be limited to a concise single sentence. <code>message</code> field MUST NOT be omitted. <code>message</code> field SHOULD start with a capital letter and SHOULD NOT end with a dot.
+* <code>data</code> field MUST be a valid UTF-8 string value that contains any implementation-dependent textual data (including debug traces). <code>data</code> field MAY be omitted, however it’s RECOMMENDED for servers to return it. <code>data</code> value SHOULD contain additional information about error and be aimed to clarify the error reason. <code>data</code> value MAY NOT be exactly the same across Neo node implementations.
+
+===Error codes===
+
+Compliant server implementations can use error codes defined by JSON-RPC
+specification (from -32700 to -32600) for any calls when they're
+appropriate. The most often used one is -32602 "Invalid params", it SHOULD be
```

Comment:
appropriate or inappropriate.

These means that from -32700 and -32600 are errors for invalid params, right?
It is not fully clear to me

Reply:
`appropriate`. They can and should be used the way they're specified in JSON-RPC 2.0.

Reply:
Resolved, no changes required.